### DISCUSSION TITLE
onBlur / onFocus usage with useDateField

### DISCUSSION THREAD

**Discussion by jake-chapman-mark43** [adobe/react-spectrum#4292] [Q&A]
We have someone using our DateInput and DateRate which uses the associated spectrum hooks. Spreading `{...fieldProps}` onto `DateField` which is passing all props into appropriate hooks. We've noticed that majority of default event handlers like `onBlur` `onFocus` etc all get stripped. At first I thought it was a prop spreading issue. But after some logical thinking and working on it I realized it kind of made sense since there are no "text inputs" per say, but we do have editable segments that do receive `onFocus` and `onBlur` events. I can add these manually to `<segment>` but they are fired off for each individual segment of the component being showed based on props.

They're needing to save the value to the database when the user has selected a date, what's an appropriate way to handle `onBlur` for something like DateInput/DateRange since there isn't a 'single' event for those but will be firing off multiple times?

**Comment by snowystinger** [comment]
I think `useDateField` will handle this for you? https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/datepicker/src/useDateField.ts#L66 is it not? and can you provide a codesandbox? might be easier to take a look at something reproducing it

**Reply by jake-chapman-mark43** [reply]
Don't think this is working as intended. Appears the `onBlurWithin` just isn't getting called. 

Copied over one of the examples from docs: https://codesandbox.io/s/recursing-montalcini-fkdbhf

**Reply by snowystinger** [reply]
seems like a bug, opened here https://github.com/adobe/react-spectrum/issues/4324
thanks for the sandbox!

**Reply by LFDanLu** [reply]
Looks like the packages needed to be updated, seems to work after bumping to latest: https://codesandbox.io/s/beautiful-breeze-ebvzit

**Reply by rdangol1** [reply]
Adding to this when we click directly below outside of the date field the focus moves. 
CodeSandbox: https://codesandbox.io/s/cocky-leavitt-ylsh6c?file=/src/App.js
recording:

https://github.com/adobe/react-spectrum/assets/65328698/4d02ae2c-3c73-4ff0-a26b-e6d674e8feea

**Reply by LFDanLu** [reply]
That behavior is discussed here: https://github.com/adobe/react-spectrum/issues/3164 (thanks @snowystinger for digging that up for me), looks to be related to `contentEditable`. That issue has some proposed fixes in the replies that you can check out.
